packages/seeder/index.js

```diff
@@ -1,5 +1,87 @@
-// your code starts here
+const { join } = require('path')
+const { homedir } = require('os')
+const Hyperdrive = require('@geut/hyperdrive-promise')
+const Corestore = require('corestore')
+const CSN = require('corestore-swarm-networking')
+const raf = require('random-access-file')
 
-module.exports = function seeder () {
-  console.log('seeder---')
+const DEFAULT_OPTS = {
+  announce: true,
+  lookup: true,
+  hyperdriveOpts: {
+    sparse: false,
+    latest: true
+  },
+  storageLocation: join(homedir(), 'permanent-seeder'),
+  corestoreOpts: {
+    sparse: false,
+    eagerUpdate: true
+  },
+  swarmOpts: {
+    announceLocalAddress: true
+  },
+  dhtOpts: {}
 }
+
+const getCoreStore = (storageLocation, name) => {
+  const location = join(storageLocation, name)
+  return file => raf(join(location, file))
+}
+
+class Seeder {
+  constructor (opts = {}) {
+    this.opts = { ...DEFAULT_OPTS, ...opts }
+    this.drives = new Map()
+    this.ready = false
+  }
+
+  async init () {
+    if (this.ready) return
```

Comment:
In which scenario would we call this method more than once?

Reply:
when adding more methods, I think its handy from the DX side, to hide the `init` step.